arxivdigest/api/database.py

```diff
@@ -252,6 +252,29 @@ def get_topic_recommendations(ids):
 
         return users
 
+def get_user_topics(ids):
+    """Returns all topics a user has from the user_topics table and
+    the topic_recommendations table as a dictionary 
+    with userid: [topics]."""
+    with closing(getDb().cursor(dictionary=True)) as cur:
+        placeholders = ','.join(['%s'] * len(ids))
+        sql = '''SELECT temp.user_id, t.topic FROM                
+            (SELECT tr.user_id, tr.topic_id FROM  topic_recommendations tr
+            LEFT JOIN user_topics ut ON ut.topic_id = tr.topic_id and 
+            tr.user_id = ut.user_id WHERE tr.user_id in (%s)
+            UNION
+            SELECT ut.user_id, ut.topic_id FROM  topic_recommendations tr
+            RIGHT JOIN user_topics ut ON ut.topic_id = tr.topic_id and 
+            tr.user_id = ut.user_id WHERE tr.user_id in (%s)) temp
+            LEFT JOIN topics t 
+            on t.topic_id=temp.topic_id''' % (placeholders, placeholders)
```

Comment:
Should only select topics where interleaving order is not null or is user_added or user_rejected.

Several systems can recommend the same topics until the user has interacted with it.